mono/metadata/w32file-unix.c

```diff
@@ -2487,7 +2487,7 @@ mono_w32file_close (gpointer handle)
 	/* FIXME: we transition here and not in file_close, pipe_close,
 	 * console_close because w32handle_close is not coop aware yet, but it
 	 * calls back into w32file. */
-	res = mono_w32handle_close (handle);
+	res = mono_w32handle_close (handle, MONO_W32HANDLE_FILE | MONO_W32HANDLE_CONSOLE);
```

Comment:
`MONO_W32HANDLE_*`  are not flags, so `MONO_W32HANDLE_FILE | MONO_W32HANDLE_CONSOLE` doesn't have any meaning. You should change their definition in w32handle.h to be 
```
typedef enum {
	MONO_W32HANDLE_UNUSED = 0,
	MONO_W32HANDLE_FILE = 1 << 0,
	MONO_W32HANDLE_CONSOLE = 1 << 1,
	MONO_W32HANDLE_THREAD = 1 << 2, // this should be removed since it's not in use anymore
	MONO_W32HANDLE_SEM = 1 << 3,
	MONO_W32HANDLE_MUTEX = 1 << 4,
	MONO_W32HANDLE_EVENT = 1 << 5,
	MONO_W32HANDLE_SOCKET = 1 << 6,
	MONO_W32HANDLE_FIND = 1 << 7,
	MONO_W32HANDLE_PROCESS = 1 << 8,
	MONO_W32HANDLE_PIPE = 1 << 9,
	MONO_W32HANDLE_NAMEDMUTEX = 1 << 10,
	MONO_W32HANDLE_NAMEDSEM = 1 << 11,
	MONO_W32HANDLE_NAMEDEVENT = 1 << 12,
	MONO_W32HANDLE_COUNT
} MonoW32HandleType;
```